### PR TITLE
Allowed to get associated fields when returning scalar from api.

### PR DESCRIPTION
**Pull request by Daniel-KM**
When using the api search, it's not possible to return the scalar values for the foreign key of a table. For example, it's not possible to get a list of user ids of items.

### CODE REVIEW

File: application/src/Api/Adapter/AbstractEntityAdapter.php [hunk 1]
```diff
@@ -235,14 +235,21 @@ public function search(Request $request)
...
-                    $scalarField, $entityClass
-                ));
+                $associationNames = $classMetadata->getAssociationNames();
+                if (!in_array($scalarField, $associationNames)) {
+                    throw new Exception\BadRequestException(sprintf(
+                        $this->getTranslator()->translate('The "%1$s" field is not available in the %2$s entity class.'),
+                        $scalarField, $entityClass
+                    ));
+                }
+                $qb->join('omeka_root.' . $scalarField, $scalarField . '_');
```
Comment (zerocrates) [inline application/src/Api/Adapter/AbstractEntityAdapter.php:248]: I think it's better off using `createAlias()` for this join's alias as is usual for other joins.

**Comment by zerocrates** [comment]
To my reading this _only_ works on the ID column of associated entities, correct?

I'm wondering if we can do this with the `IDENTITY()` DQL function and avoid the join. It just would require finding only the associations with the correct "owning side," I think.

**Comment by Daniel-KM** [comment]
Indeed, the join is automatically added by doctrine itself with identity().

**Comment by Daniel-KM** [comment]
Yes, the it's only for related entity ids, other data are available with the getFieldNames().

**Pull request merged**